pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -1017,20 +983,17 @@ int kprobe__udp_destroy_sock(struct pt_regs* ctx) {
 
 //region sys_enter_bind
 
-static __always_inline int sys_enter_bind(__u64 fd, struct sockaddr* addr) {
+static __always_inline int sys_enter_bind(struct socket* sock, struct sockaddr* addr) {
     __u64 tid = bpf_get_current_pid_tgid();
 
-    // determine if the fd for this process is an unbound UDP socket
-    __u64 fd_and_tid = (tid << 32) | fd;
-    __u64* u = bpf_map_lookup_elem(&unbound_sockets, &fd_and_tid);
-
-    if (u == NULL) {
-        log_debug("sys_enter_bind: bind happened, but not on a UDP socket, fd=%u, tid=%u\n", fd, tid);
+    __u16 type = 0;
+    bpf_probe_read(&type, sizeof(__u16), &sock->type);
```

Comment:
Does this need runtime compilation?

Reply:
It looked stable from `3.10` onwards. There is only an enum before `type`.

Reply:
Not future proof though right? It is probably still low risk though. We could wait on rolling this out till after runtime compilation is rolled out.

Reply:
Technically, nothing about a kprobe is future-proof because they can change the function names/arguments at any time. Tracepoints/syscalls are the most stable, but couldn't be used for arm64 as stated above.

I think the risk of this changing is extremely low. I haven't seen official guidance, but it seems that kernel devs tend to add to the end of structures, unless there is a good reason. I don't think it needs to wait for runtime compilation.

Reply:
Ok, that makes sense.